modules/file-system/filestore/main.tf

```diff
@@ -104,5 +104,10 @@ resource "google_filestore_instance" "filestore_instance" {
         BASIC_HDD or BASIC_SSD tiers. Otherwise the range size must be 24.
         EOT
     }
+
+    precondition {
+      condition     = substr(var.filestore_tier, 0, 5) != "BASIC" || var.protocol != "NFS_V4_1"
```

Comment:
```suggestion
      condition     = !startswith(var.filestore_tier, "BASIC") || var.protocol != "NFS_V4_1"
```

Reply:
1. Double check my logic, please. :)
2. Bump required_version in versions.tf to `>= 1.3.0`